### ISSUE TITLE
Should nwg-look be able to set its own theme?

### ISSUE THREAD

**Issue by simonm** [nwg-piotr/nwg-look#66]
Hello

I'm going through the whole gtk-2, 3, 4, xsettings theming shenanigans for Hyprland.  

I have two questions:

1. Should the theme I set in `nwg-look `be applied next time I open `nwg-look`, or even as I set and apply the theme in the app?  
2. Should the themes selected in the "Widgets" pane show a preview in` nwg-look`?  

Neither of these things happen.

- The application` lxappearance` honors the theme choice I make in `nwg-look`. nwg-look does not.
- The application `lxappearance` shows dynamic widget style previews in the widget tab when I select a new style. It also updates its own theme when I click apply. 

I am reviewing the open issues and see the discussion about gtk4 and [libadwaita-without-adwaita-git](https://aur.archlinux.org/packages/libadwaita-without-adwaita-git/)AUR. I have installed libadwaita without libadwaita.

My apologies if this is a bad question - I've been looking into this for a while and have not been able to make much progress.

I am on Hyprland. I have also tried setting all of this using env vars. The results are inconsistent which is why I wanted to get nwg-look to work. It seems to be a popular solution for GTK theming!

Let me know if I can provide additional information.

```
❯ nwg-look -v
nwg-look version 0.2.7
```

```
❯ expac '%-30n %v' -s 'nwg'
gopsuinfo                      0.1.6-1
nwg-bar-bin                    0.1.6-2
nwg-clipman                    0.2.2-1
nwg-displays                   0.3.20-1
nwg-dock-bin                   0.4.0-1
nwg-dock-hyprland-bin          0.2.0-1
nwg-drawer-bin                 0.4.9-1
nwg-icon-picker                0.1.1-3
nwg-look                       0.2.7-1
nwg-menu-bin                   0.1.3-2
nwg-panel                      0.9.37-1
nwg-readme-browser             0.1.5-2
nwg-shell                      0.5.37-1
nwg-shell-config               0.5.42-1
nwg-shell-wallpapers           1.5-1
```

And maybe this is helpful:

```
❯ expac '%-30n %v' -s 'Hyprland'
hyprcursor                     0.1.9-1
hyprevents-git                 11.09b54e7-1
hypridle                       0.1.2-1
hyprland                       0.41.2-3
hyprlock                       0.4.1-1
hyprmixer                      1.0.1-1
hyprnome                       0.3.0-1
hyprpanel                      0.1.10-1
hyprprop-git                   16.46d12db-1
hyprshade                      3.2.1-1
hyprshot                       1.3.0-1
hyprutils                      0.2.0-1
hyprwayland-scanner-git        0.3.10.r7.g55c78a56-1
nwg-displays                   0.3.20-1
nwg-dock-hyprland-bin          0.2.0-1
nwg-panel                      0.9.37-1
xdg-desktop-portal-hyprland    1.3.3-2
```

Again, apologies if I have misunderstood anything.

Thank you.

**Comment by nwg-piotr** [comment]
Hi Simon,

> Should the theme I set in nwg-look be applied next time I open nwg-look, or even as I set and apply the theme in the app?

Yes.

> Should the themes selected in the "Widgets" pane show a preview in nwg-look?

Yes.

Remove all the theming-related env vars, as they override gsettings on Hyprland. I don't know which Hyprland version you're on. I only tested up to  v0.41.2.

**Comment by simonm** [comment]
Hi Piotr

Many thanks for the quick response! This is helpful.

I will close this ticket and continue my epic quest...

Hope you have a lovely day.

**Comment by nwg-piotr** [comment]
If you come up with something revolutionary, let me know. I'm going to spend the next week drinking beer and fishing, but I'll get to that when I'm back.

**Comment by simonm** [comment]
Haha! just saw your last comment. 
Hope you have a fantastic break and that the beer is cold and the fish are biting.

For the record, in case this might be helpful to anyone else: I had conflicting configs and environment variables floating around. I removed a bunch of themes, removed theme-related config from all of my dotfiles and started again.

My solution was to put my chosen theme and icons in `~/.themes` and `~/.icons`, select them with `nwg-panel` and then `apply` to write the correct configs (`~/.config/gtk*` `~/.config/xsettingsd` etc.). I then added these directories to my .dotfiles repo and so can replicate my theme choices across my different systems (laptop, workstation etc).

All good now!

Thanks again for your fantastic work on all of the nwg- apps. Really helpful.
